tests/test_cobs.cc

```diff
@@ -0,0 +1,146 @@
+#include "cobs.h"
+#include <cstring>
+
+/*
+Throw exceptions so we can catch them with gdb (catch catch).
+*/
+void throw_assert(bool condition, const char * const remark) {
+    if (condition == false) {
+        throw remark;
+    }
+}
+
+/*
+Fill a byte buffer with a given value.
+*/
+void fill(uint8_t * const b, size_t l, uint8_t v) {
+    std::memset(b, v, l);
+}
+
+/*
+Check that a byte buffer contains only a given value.
+*/
+void mem_is(uint8_t * const b, size_t l, uint8_t v) {
+    for (size_t i = 0; i < l; i++) {
+        throw_assert(b[i] == v, "");
+    }
+}
+
+int main() {
+    const uint8_t X = 0xdd; // non-zero payload
+    const uint8_t F = 0xcc; // fill
+
+    // Encode the empty packet.
+    {
+        uint8_t b1[1000]; fill(b1, sizeof(b1), X);
+        uint8_t b2[1000]; fill(b2, sizeof(b2), F);
+        uint8_t b3[1000]; fill(b3, sizeof(b3), F);
+        const size_t b1_len = 0;
+        const size_t b2_len = cobs::encode(b1, b1_len, b2, sizeof(b2));
+        mem_is(&b2[0],   1, 0);
+        mem_is(&b2[1], 999, F);
```

Comment:
seems like ```sizeof(b2) - b2_len``` would be better here in case the buffer size is changed in the future.